### PR TITLE
ref: Remove duplicated projects from organization details

### PR DESCRIPTION
**Pull request by dcramer**
This refactors a large amount of UI components to stop using `team.projects` and instead rely on the `project.teams` association to identify which teams are associated to which projects.

The primary initial motivator here is the huge performance hit both in response payload but also in serialization.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/team/expandedTeamList.jsx [hunk 1]
```diff
@@ -1,231 +0,0 @@
-import {Link} from 'react-router';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/settings/team/expandedTeamList.jsx]: this file was not referenced anywhere

**Comment by dcramer** [comment]
@macqueen this is what i was talking about

File: src/sentry/api/bases/organization.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def has_object_permission(self, request, view, organization):
             )
 
         elif request.auth:
-            return request.auth.organization_id == organization.id
```
Comment (dcramer) [inline src/sentry/api/bases/organization.py]: this isnt heavily used yet but its going to move us towards using a single source for 'what memberships and permissions are available to me'

File: src/sentry/api/bases/team.py [hunk 3]
```diff
@@ -9,9 +9,6 @@
 
 
 def has_team_permission(request, team, scope_map):
-    if not (request.user and request.user.is_authenticated()) and request.auth:
```
Comment (dcramer) [inline src/sentry/api/bases/team.py]: dead per the org base change

File: src/sentry/static/sentry/app/components/actionOverlay.jsx [hunk 4]
```diff
@@ -1,77 +0,0 @@
-import PropTypes from 'prop-types';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/actionOverlay.jsx]: dead code

File: src/sentry/static/sentry/app/components/requiredAdminActions.jsx [hunk 5]
```diff
@@ -1,11 +0,0 @@
-import SetCallsignsAction from '../views/requiredAdminActions/setCallsigns';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/requiredAdminActions.jsx]: dead code

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 6]
```diff
@@ -10,7 +10,6 @@ import {load as loadIncidents} from '../../actionCreators/incidents';
 import Broadcasts from './broadcasts';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: changes are dead code

File: src/sentry/static/sentry/app/routes.jsx [hunk 7]
```diff
@@ -91,7 +91,6 @@ import ReleaseDetails from './views/releaseDetails';
 import ReleaseNewEvents from './views/releaseNewEvents';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/routes.jsx]: changes are dead code

File: src/sentry/static/sentry/app/routes.jsx [hunk 8]
```diff
@@ -360,8 +359,8 @@ const projectSettingsRoutes = (
       name="Integration Details"
       component={errorHandler(ProjectPluginDetails)}
     />
-    // XXX(epurkhiser): This lives under project configurations for now until
```
Comment (dcramer) [inline src/sentry/static/sentry/app/routes.jsx]: this seems broken -- im removing the comment

File: src/sentry/static/sentry/app/views/organizationContext.jsx [hunk 9]
```diff
@@ -22,10 +22,6 @@ let ERROR_TYPES = {
   ORG_NOT_FOUND: 'ORG_NOT_FOUND',
 };
 
-function getRequiredAdminActions(org) {
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationContext.jsx]: changes are dead code

File: src/sentry/static/sentry/app/views/projectChooser.jsx [hunk 10]
```diff
@@ -22,8 +22,7 @@ const ProjectChooser = createReactClass({
 
   redirectNoMultipleProjects() {
     let org = this.getOrganization();
-    let teams = org.teams.filter(team => team.projects.length > 0);
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/projectChooser.jsx]: this changes the behavior and removes teams entirely from the display

File: src/sentry/static/sentry/app/views/requiredAdminActions/setCallsigns.jsx [hunk 11]
```diff
@@ -1,217 +0,0 @@
-import React from 'react';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/requiredAdminActions/setCallsigns.jsx]: dead code

**Comment by dcramer** [comment]
I went ahead and removed the rest of the callsign legacy code.

**Comment by dcramer** [comment]
There should still be one failing test that I've asked @billyvg to help with

**Comment by dcramer** [comment]
```
 FAIL  tests/js/spec/views/organizationTeamProjects.spec.jsx
  ● Console

    console.error src/sentry/static/sentry/app/__mocks__/api.jsx:83
      No mocked response found for request. /projects/org-slug/project-slug/teams/team-slug/ DELETE
    console.error node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/virtual-console.js:29
      Error: Uncaught [Error: expect(jest.fn()).toHaveBeenCalledTimes(1)

      Expected mock function to have been called one time, but it was called zero times.]
          at reportException (/Users/dcramer/Development/sentry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:66:24)
          at Timeout.callback [as _onTimeout] (/Users/dcramer/Development/sentry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/browser/Window.js:594:7)
          at ontimeout (timers.js:475:11)
          at tryOnTimeout (timers.js:310:5)
          at Timer.listOnTimeout (timers.js:270:5) { Error: expect(jest.fn()).toHaveBeenCalledTimes(1)

      Expected mock function to have been called one time, but it was called zero times.
          at /Users/dcramer/Development/sentry/tests/js/spec/views/organizationTeamProjects.spec.jsx:102:26
          at Timeout.callback [as _onTimeout] (/Users/dcramer/Development/sentry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/browser/Window.js:592:19)
          at ontimeout (timers.js:475:11)
          at tryOnTimeout (timers.js:310:5)
          at Timer.listOnTimeout (timers.js:270:5) matcherResult: { message: [Function], pass: false } }

  ● OrganizationTeamProjects › Should allow adding and removing projects

    expect(jest.fn()).toHaveBeenCalledTimes(1)

    Expected mock function to have been called one time, but it was called zero times.

      100 |       remove.simulate('click');
      101 |
    > 102 |       expect(deleteMock).toHaveBeenCalledTimes(1);
      103 |
      104 |       done();
      105 |     }, 1);

      at tests/js/spec/views/organizationTeamProjects.spec.jsx:102:26
```

**Comment by billyvg** [comment]
`noteInput`, `ownerInput`, `assigneeSelector` rely on TeamStore.`projects`.

**Review by billyvg — APPROVED** [review]
I haven't reviewed the python but the JS looks good.

**Comment by dcramer** [comment]
might have broken team name in org sidebar widget (for project list), but ill look after merge

**Pull request merged**